app/components/data-download.tsx

```diff
@@ -98,43 +62,162 @@ const DataDownloadInner = memo(
             : acc;
         }, {})
       );
-    }, [columns, observations]);
+      const columnKeys = Object.keys(columns).map((d) => columns[d].label);
+      return { data, columnKeys };
+    } else {
+      return { data: [], columnKeys: [] };
+    }
+  }, [fetchedData?.dataCubeByIri]);
 
-    const downloadData = useCallback(() => {
-      fetch("/api/download", {
-        method: "POST",
-        headers: {
-          "Content-Type": "application/json",
-        },
-        body: JSON.stringify({
-          columnKeys: Object.keys(columns).map((d) => columns[d].label),
-          data,
-          fileFormat,
-        }),
-      }).then((res) =>
-        res.blob().then((blob) => saveAs(blob, `${title}.${fileFormat}`))
-      );
-    }, [columns, data, fileFormat, title]);
+  return { data, columnKeys };
+};
+
+export const DataDownloadMenu = memo(
+  ({
+    dataSetIri,
+    title,
+    chartConfig,
+  }: {
+    dataSetIri: string;
+    title: string;
+    chartConfig: ChartConfig;
+  }) => {
+    const allPrepared = usePreparedData({
+      dataSetIri,
+      chartConfig,
+      extent: "all",
+    });
+    const visiblePrepared = usePreparedData({
+      dataSetIri,
+      chartConfig,
+      extent: "visible",
+    });
+    const popupState = usePopupState({
+      variant: "popover",
+      popupId: "dataDownloadMenu",
+    });
 
     return (
-      <DownloadButton onClick={() => downloadData()}>
-        {extent === "visible" ? (
-          <Trans id="button.download.data.visible">Download visible data</Trans>
-        ) : (
-          <Trans id="button.download.data.all">Download all data</Trans>
-        )}
-      </DownloadButton>
+      <>
+        <DownloadButton iconName="download" {...bindHover(popupState)}>
+          <Trans id="button.download">Download</Trans>
+        </DownloadButton>
+        <HoverMenu
+          {...bindMenu(popupState)}
+          anchorOrigin={{ vertical: "top", horizontal: "center" }}
+          transformOrigin={{ vertical: "bottom", horizontal: "center" }}
+        >
+          <Typography
+            sx={{
+              ...PADDING_PROPS,
```

Comment:
This is just a placeholder text (menu title), so I am not sure it should have MenuItem appearance (mostly onHover styling I think) - it looks like it can be selected then. I needed the same padding for both download options & the title, so that's why there is this prop. Do you see a better approach here?

Reply:
Either have a MenuTitle component, or maybe use the subheader prop on a MenuList component

Reply:
I'd use the menuListProps={{ subheader: 'Download' }} to be able to style the subheader inside the theme. Weirdly by the subheader is not themed.

Reply:
If not using the theme, I think it would be clearer with

```
const MenuTitle = styled(Typography)({ ...PADDING_PROPS, borderBottom: '1px solid', borderBottomColor: 'grey.500' })
```

The disadvantage here would be that styling would not be in the theme.